### PR TITLE
Move Block Operation Length Checks to ProcessOperations

### PR DESCRIPTION
**Pull request by 0xKiwi**
Brought up in #2892, this moves all length checks to a `verifyOperationLengths` function to prevent extra unneeded computing if an invalid block is submitted.

### CODE REVIEW

File: beacon-chain/core/state/transition_test.go [hunk 1]
```diff
@@ -246,131 +246,6 @@ func TestProcessBlock_IncorrectProcessBlockAttestations(t *testing.T) {
 	}
 }
 
-func TestProcessBlock_IncorrectProcessExits(t *testing.T) {
```
Comment (terencechain) [inline beacon-chain/core/state/transition_test.go]: why remove this?

File: beacon-chain/core/state/transition.go [hunk 2]
```diff
@@ -255,6 +255,46 @@ func ProcessOperations(
 	ctx, span := trace.StartSpan(ctx, "beacon-chain.ChainService.state.ProcessOperations")
 	defer span.End()
 
+	if uint64(len(body.ProposerSlashings)) > params.BeaconConfig().MaxProposerSlashings {
```
Comment (terencechain) [inline beacon-chain/core/state/transition.go]: Better to refactor these to a separate function? ex: `VerifyOperationLength`?

File: beacon-chain/core/state/transition_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xKiwi) [reply beacon-chain/core/state/transition_test.go]: It's testing for a block that contains over the max of voluntary exits, it doesn't need to be so complicated anymore.

File: beacon-chain/core/state/transition.go (hunk 2 above)
Comment (0xKiwi) [reply beacon-chain/core/state/transition.go]: Good idea!

File: beacon-chain/core/state/transition_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (terencechain) [reply beacon-chain/core/state/transition_test.go]: sounds good

**Pull request merged**